### PR TITLE
elf: contemplate more patching scenarios

### PR DESCRIPTION
**Pull request by sergiusens**
Includes support for building classic confined snaps on release like
bionic that target a 16 base and also adds support for using hooks built
with snapcraft.

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

**Comment by sergiusens** [comment]
For this I tested installing and building the following on bionic
- rust-hello (classic)
- hello-classic
- pyhooks

File: snapcraft/_options.py [hunk 1]
```diff
@@ -223,6 +224,16 @@ def __init__(self, use_geoip=False, parallel_builds=True,
...
 
+    @property
+    def is_host_comatible_with_base(self):
+        codename = None
+        with suppress(errors.OsReleaseCodenameError):
+            codename = OsRelease().version_codename()
+            logger.debug('Running on {!r}'.format(codename))
+
+        # TODO support more bases
+        return codename == 'xenial' or codename == 'trusty'
```
Comment (kyrofa) [inline snapcraft/_options.py]: Ah, good call adding Trusty, forgot about that one.

**Comment by sergiusens** [comment]
Great, give me a nod when you are ready

File: snapcraft/_options.py (hunk 1 above)
Comment (sergiusens) [reply snapcraft/_options.py]: Yeah, I change the name of this `property` to reflect that.

**Comment by sergiusens** [comment]
All greens! 😄  I will update the branch now

**Comment by elopio** [comment]
snapcraft/internal/pluginhandler/__init__.py:527: error: Argument 1 to "_handle_dependencies" of "PluginHandler" has incompatible type "Set[str]"; expected "Set[ElfFile]"

File: snapcraft/_options.py [hunk 2]
```diff
@@ -223,6 +224,16 @@ def __init__(self, use_geoip=False, parallel_builds=True,
         self._set_machine(target_deb_arch)
         self.__debug = debug
 
+    @property
+    def is_host_comatible_with_base(self):
```
Comment (elopio) [inline snapcraft/_options.py]: typo: s/comatible/compatible

**Pull request merged**